### PR TITLE
Improve @Deprecated Enum Type developer experience

### PR DESCRIPTION
**Pull request by tombailey14**
## Description

This adds the feature mentioned in issue #7627 in which the deprecated directive comment was not being added to the generated `resolver-types.ts` file for  the `ENUM_VALUE` type but was for the `FIELD_DEFINITION` type. This PR uses the same function but makes it usable for both deprecated directive types.

Related #7627 

## Type of change

- [X] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

A test has been added in the `typescript.spec.ts` file testing that the correct comment is generated. Following a similar structure to the `FIELD_DEFINITION` test already implemented.

Additionally a screenshot below showing the deprecated value being picked up by the IDE and a line through it.
Screenshot 2022-03-07 at 19 39 20

## Checklist:

- [X] I have followed the [CONTRIBUTING](https://github.com/the-guild-org/Stack/blob/master/CONTRIBUTING.md) doc and the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/plugins/other/visitor-plugin-common/src/base-types-visitor.ts [hunk 1]
```diff
@@ -456,7 +456,7 @@ export class BaseTypesVisitor<
   FieldDefinition(node: FieldDefinitionNode): string {
     const typeString = node.type as any as string;
     const { type } = this._parsedConfig.declarationKind;
-    const comment = this.getFieldComment(node);
+    const comment = this.getDeprecationComment(node);
```
Comment (charlypoly) [inline packages/plugins/other/visitor-plugin-common/src/base-types-visitor.ts:459]: `getDeprecationComment(node)` is still handling some logic related to comments in general, not only deprecation.

Please leave it as `getNodeComment()` 🙂 

```suggestion
    const comment = this.getNodeComment(node);
```

File: packages/plugins/other/visitor-plugin-common/src/base-types-visitor.ts [hunk 2]
```diff
@@ -799,6 +788,17 @@ export class BaseTypesVisitor<
     return null;
   }
 
+  getDeprecationComment(node: FieldDefinitionNode | EnumValueDefinitionNode): string {
```
Comment (charlypoly) [inline packages/plugins/other/visitor-plugin-common/src/base-types-visitor.ts:791]:
```suggestion
  getNodeComment(node: FieldDefinitionNode | EnumValueDefinitionNode): string {
```

File: packages/plugins/other/visitor-plugin-common/src/base-types-visitor.ts [hunk 3]
```diff
@@ -664,7 +653,7 @@ export class BaseTypesVisitor<
         const optionName = this.makeValidEnumIdentifier(
           this.convertName(enumOption, { useTypesPrefix: false, transformUnderscore: true })
         );
-        const comment = transformComment(enumOption.description as any as string, 1);
+        const comment = this.getDeprecationComment(enumOption);
```
Comment (charlypoly) [inline packages/plugins/other/visitor-plugin-common/src/base-types-visitor.ts:656]:
```suggestion
        const comment = this.getNodeComment(enumOption);
```

File: packages/plugins/typescript/typescript/src/visitor.ts [hunk 4]
```diff
@@ -258,7 +258,7 @@ export class TsVisitor<
       : (node.type as any as string);
     const originalFieldNode = parent[key] as FieldDefinitionNode;
     const addOptionalSign = !this.config.avoidOptionals.field && originalFieldNode.type.kind !== Kind.NON_NULL_TYPE;
-    const comment = this.getFieldComment(node);
+    const comment = this.getDeprecationComment(node);
```
Comment (charlypoly) [inline packages/plugins/typescript/typescript/src/visitor.ts:261]:
```suggestion
    const comment = this.getNodeComment(node);
```

**Review by charlypoly — CHANGES_REQUESTED** [review]
It looks good, thanks!

Some minor naming changes and we're good to ship 📦

**Comment by tombailey14** [comment]
That should be it updated now! Thanks for the review @charlypoly

**Review by charlypoly — APPROVED** [review]

**Pull request merged**